### ISSUE TITLE
What is equivalent of random crop function from pytorch?

### ISSUE THREAD

**Issue by hadaev8** [NVIDIA/DALI#4801]
### Describe the question.

Im trying to redo my data loader from pytorch, it load pair of images, cat by channel axis, crop and downscale.

For now i have following dali implementation and stuck with random crop, i have images of size 4-8k, and i want to crop regions of 1k without any resizing, with random_aspect_ratio=(1.0, 1.0) and random_area=(1.0, 1.0) it just downscales it without cropping, how should i approach my task?

```
class GanDataset(torch.utils.data.Dataset):
    def __init__(self, hp):
        self.totensor = transforms.ToTensor()
        self.files = []

        files = glob.glob(os.path.normpath(hp.folder + '/**/*Albedo*'), recursive=True)
        for file in files:
            if os.path.exists(file.replace('Albedo', 'Normal')):
                self.files.append([file, file.replace('Albedo', 'Normal')])
        self.transforms = transforms.Compose([transforms.RandomCrop(hp.im_size),
                                              transforms.RandomHorizontalFlip(),
                                              transforms.RandomVerticalFlip()])
        self.downscale = transforms.Compose([transforms.Resize(hp.im_size//4, antialias=False)])

    def load_image(self, file):
        diffuse = self.totensor(Image.open(file[0]))
        normal = self.totensor(Image.open(file[1]))
        image = torch.cat([diffuse, normal], dim=0)
        image = self.transforms(image)
        image = image * 2 - 1
        return image

    def __getitem__(self, index):
        img = self.load_image(self.files[index])
        return img, self.downscale(img)

    def __len__(self):
        return len(self.files)


def get_dataloader(hp, inf=True):
    trainset = GanDataset(hp)
    if inf:
        sampler = torch.utils.data.RandomSampler(trainset, replacement=True, num_samples=hp.num_samples)
    else:
        sampler = None
    train_loader = torch.utils.data.DataLoader(trainset, num_workers=hp.num_workers,
                                               sampler=sampler, batch_size=hp.batch_size,
                                               pin_memory=True, drop_last=True)
    return train_loader
```

```
class ExternalInputIterator(object):
    def __init__(self, batch_size):
        self.images_dir = image_dir
        self.batch_size = batch_size
        files = glob.glob(os.path.normpath(image_dir + '/**/*4K*Albedo*.jpg'), recursive=True)
        self.files = []
        for file in files:
            if os.path.exists(file.replace('Albedo', 'Normal')):
                self.files.append([file, file.replace('Albedo', 'Normal')])
        self.n = len(self.files)

    def __iter__(self):
        self.i = 0
        shuffle(self.files)
        return self

    def __next__(self):
        batch0 = []
        batch1 = []
        if self.i >= self.n:
            self.__iter__()
            raise StopIteration
        for _ in range(self.batch_size):
            fname0, fname1 = self.files[self.i]
            batch0.append(np.fromfile(fname0, dtype=np.uint8))
            batch1.append(np.fromfile(fname1, dtype=np.uint8))
            self.i += 1
        return (batch0, batch1)

    def __len__(self):
        return self.n

    next = __next__

# eii = ExternalInputIterator(batch_size)
def ExternalSourcePipeline(batch_size, num_threads, device_id, external_data):
    pipe = Pipeline(batch_size=batch_size, num_threads=num_threads, device_id=device_id)
    with pipe:
        jpegs0, jpegs1 = fn.external_source(source=external_data, num_outputs=2, dtype=types.UINT8)
        decode0 = fn.decoders.image(jpegs0, device="mixed", output_type=types.RGB)
        # decode0 = fn.random_resized_crop(decode0, size=(1024, 1024), random_aspect_ratio=(1.0, 1.0))
        decode1 = fn.decoders.image(jpegs1, device="mixed", output_type=types.RGB)
        # decode1 = fn.random_resized_crop(decode1, size=(1024, 1024), random_aspect_ratio=(1.0, 1.0))
        decode = fn.cat(decode0, decode1, axis=-1)
        decode1 = fn.decoders.image_random_crop(decode, size=(1024, 1024), random_aspect_ratio=(1.0, 1.0), random_area=(1.0, 1.0))
        pipe.set_outputs(decode, decode1)
    return pipe
# pipe.build()

from nvidia.dali.plugin.pytorch import DALIGenericIterator
from nvidia.dali.plugin.pytorch import LastBatchPolicy

eii = ExternalInputIterator(batch_size)
pipe = ExternalSourcePipeline(batch_size=batch_size, num_threads=2, device_id = 0,
                              external_data = eii)
pii = DALIGenericIterator(pipe, output_map=['data', 'data1'], last_batch_padded=True, last_batch_policy=LastBatchPolicy.PARTIAL)

for i, data in enumerate(pii):
    break
```

### Check for duplicates

- [X] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

**Comment by hadaev8** [comment]
Seem this is what i need, thanks.
